### PR TITLE
build: fix opkg flags in rootfs

### LINKED ISSUE #14427: Incorrect opkg flags in rootfs image
### Describe the bug

When creating the rootfs image, the default packages are installed using `opkg` which [adds](https://github.com/openwrt/opkg-lede/blob/d038e5b6d155784575f62a66a8bb7e874173e92e/libopkg/opkg_install.c#L1198) the `user` flag for most packages[^1].

The `user` flag  should only be applied to packages have been installed by the user and thus should not be applied to the default packages that are part of the system.

For example here are the initial package flags (truncated) from a new SNAPSHOT image on a NanoPi R4S:
```
root@OpenWrt:~# opkg status | awk '/Package:|Status:|^$/1'
Package: libc
Status: install hold installed

Package: mkf2fs
Status: install user installed

Package: opkg
Status: install user installed
...
Package: ppp-mod-pppoe
Status: install user installed
```

[^1]: Packages, like `libc`, which [specify flags](https://github.com/openwrt/openwrt/blob/90644fb2d004112c3d877abd94cd5528d7587ee1/package/libs/toolchain/Makefile#L15), like `hold`, do not get the `user` flag added by default.

### OpenWrt version

r24855-3891355015

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

rockchip/armv8

### Device

FriendlyElec NanoPi R4S

### Image kind

Official downloaded image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

### PR DESCRIPTION
**Pull request by jck112**
By default opkg sets the "user" flag when a package is installed, which resulted in most packages in the rootfs having this flag set incorrectly. This patch removes the "user" flag from all installed packages when preparing the rootfs image.

Fixes: #14427

### CODE REVIEW

**Comment by jck112** [comment]
Output of `opkg status | awk '/Package:|Status:|^$/1'` before the change (truncated for brevity):
```
root@OpenWrt:~# opkg status | awk '/Package:|Status:|^$/1'
Package: libc
Status: install hold installed

Package: mkf2fs
Status: install user installed

Package: opkg
Status: install user installed
...
Package: ppp-mod-pppoe
Status: install user installed
```

Output of `opkg status | awk '/Package:|Status:|^$/1'` after the change (truncated for brevity):
```
root@OpenWrt:~# opkg status | awk '/Package:|Status:|^$/1'
Package: libc
Status: install hold installed

Package: mkf2fs
Status: install ok installed

Package: opkg
Status: install ok installed
...
Package: ppp-mod-pppoe
Status: install ok installed
```

Here is the full contents of `/usr/lib/opkg/status` [before the change](https://github.com/openwrt/openwrt/files/13980105/status-before.txt) and [after the change](https://github.com/openwrt/openwrt/files/13980105/status-after.txt) as well as a [diff of the two](https://github.com/openwrt/openwrt/files/13980107/status-diff.txt).

File: include/rootfs.mk [hunk 1]
```diff
@@ -87,6 +87,11 @@ define prepare_rootfs
 			fi; \
 		done || true \
 	)
+	gawk -i inplace \
```
Comment (ynezz) [inline include/rootfs.mk:90]:
```suggestion
	awk -i inplace \
```

According to the https://github.com/openwrt/openwrt/blob/main/include/prereq-build.mk#L136-L138 I would say, that this needs to be `awk`.

**Review by ynezz — CHANGES_REQUESTED** [review]

File: include/rootfs.mk (hunk 1 above)
Comment (jck112) [reply include/rootfs.mk]: Changed to `awk`.

**Comment by robimarko** [comment]
@ynezz Are the changes OK?

**Comment by aparcar** [comment]
Looks great to me

**Comment by ynezz** [comment]
> @ynezz Are the changes OK?

Yes, they look OK, it makes sense and it looks, that this has been tested extensively.

**Review by ynezz — APPROVED** [review]

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Comment by Kernalapp** [comment]
@ynezz @jck112 
hello,will this commit be synced to the source repository(https://git.openwrt.org/?p=project/opkg-lede.git;a=summary)?

**Comment by Kernalapp** [comment]
> There is a bit of misunderstanding, this changes are unrelated to opkg-lede project, so if you're probably talking about the [other contribution](https://github.com/openwrt/opkg-lede/pull/1) from Justin, then this one wasn't still applied.
> 
> It probably needs to be tested within ImageBuilder as well, so we're sure, that it doesn't [introduce regressions](https://github.com/openwrt/openwrt/commit/c42b915af0ea53bc1b65bc72afc9875da08b14d2) (or any other unexpected side-effects) like this one did.

Hello, we have observed that opkg(https://git.openwrt.org/?p=project/opkg-lede.git;a=summary) has not changed for two years (no issues and commits), has opkg become inactive and declined?

**Comment by ynezz** [comment]
That is a good observation, indeed, nobody wants to touch this anymore. We believe, that https://github.com/openwrt/openwrt/pull/4294 is the right way forward.

**Comment by Kernalapp** [comment]
@ynezz Hello,I'm a Huawei engineer. I'm currently evaluating whether opkg is active. I see that there are two commits here(https://github.com/openwrt/opkg-lede/pulls), neither of which is currently incorporated. Is opkg no longer active?

**Pull request merged**